lib/libsmb2.c

```diff
@@ -3206,12 +3206,10 @@ smb2_query_directory_request_cb(struct smb2_server *server, struct smb2_context
                         pdu = smb2_cmd_query_directory_reply_async(smb2, req, &rep, NULL, cb_data);
                 }
         }
+        if (req->name) {
+                smb2_free_data(smb2, discard_const(req->name));
+        }
         if (pdu != NULL) {
```

Comment:
CHANGE Note: the comment was wrong.. this needs freeing always.  I had patched this to work around a use-after-free issue elsewhere, but that is the fault of elsewhere, not this lib.